include/deal.II/base/vectorization.h

```diff
@@ -167,9 +167,35 @@ struct EnableIfScalar<VectorizedArray<Number>>
  * similar to std::vector otherwise but always makes sure that data is
  * correctly aligned.
  *
+ * The user can explicitly control the used instruction set architecture (ISA)
+ *  extension by specifying the vector width/number of lanes via  the second
+ * template parameter of this wrapper class. For example on Intel Skylake,
+ * you have the following options for the data type double:
+ *  - VectorizedArray<double, 1> // no vectorization (auto-optimization)
+ *  - VectorizedArray<double, 2> // SSE2
+ *  - VectorizedArray<double, 4> // AVX2
+ *  - VectorizedArray<double, 8> // AVX-512 (default)
+ *
+ * and for Intel SandyBridge, Haswell, and Broadwell:
+ *  - VectorizedArray<double, 1> // no vectorization (auto-optimization)
+ *  - VectorizedArray<double, 2> // SSE2
+ *  - VectorizedArray<double, 4> // AVX2 (default)
+ *
```

Comment:
Please add an additional category:

```
for older x86 processors or in case no processor-specific compilation flags
were added (i.e., without `-D CMAKE_CXX_FLAGS=-march=native` or similar
flags):
- VectorizedArray<double, 1> // no vectorization (auto-optimization)
- VectorizedArray<double, 2> // SSE2
```